ion-schema/src/test/kotlin/com/amazon/ionschema/util/SchemaSymbolsUtilTest.kt

```diff
@@ -0,0 +1,147 @@
+import org.junit.jupiter.api.Test
+
+class SchemaSymbolsUtilTest {
+
+    companion object {
+        @JvmField
+        val ION_SCHEMA_2_0_SYMBOLS = setOf(
+            // Version marker
+            "\$ion_schema_2_0",
+            // Built in types
+            "any", "\$any",
+            "blob", "\$blob",
+            "bool", "\$bool",
+            "clob", "\$clob",
+            "decimal", "\$decimal",
+            "document",
+            "float", "\$float",
+            "int", "\$int",
+            "list", "\$list",
+            "lob", "\$lob",
+            "nothing",
+            "\$null",
+            "number", "\$number",
+            "sexp", "\$sexp",
+            "string", "\$string",
+            "struct", "\$struct",
+            "symbol", "\$symbol",
+            "text", "\$text",
+            "timestamp", "\$timestamp",
+            // Keywords
+            "all_of",
+            "annotations",
+            "any_of",
+            "as",
+            "byte_length",
+            "codepoint_length",
+            "container_length",
+            "contains",
+            "element",
+            "exponent",
+            "field_names",
+            "fields",
+            "id",
+            "ieee754_float",
+            "imports",
+            "name",
+            "not",
+            "occurs",
+            "one_of",
+            "ordered_elements",
+            "precision",
+            "regex",
+            "schema_footer",
+            "schema_header",
+            "timestamp_offset",
+            "timestamp_precision",
+            "type",
+            "user_reserved_fields",
+            "utf8_byte_length",
+            "valid_values",
+            // Ranges
+            "range",
+            "min",
+            "max",
+            "exclusive",
+            "optional",
+            "required",
+            // Timestamp Precisions
+            "year",
+            "month",
+            "day",
+            "minute",
+            "second",
+            "millisecond",
+            "microsecond",
+            "nanosecond",
+            // Float types
+            "binary16",
+            "binary32",
+            "binary64",
+            // Misc modifier annotations
+            "i",
+            "m",
+            "closed",
+            "distinct",
+            "required",
+            "\$null_or",
+        )
+    }
+
+    @Test
+    fun `Ion Schema 2 0 should have expected symbols`() {
+        val symbols = SchemaSymbolsUtil.getSymbolsTextForPath("./src/main/resources/ion-schema-schemas/") { it.path.contains("isl/ion_schema_2_0") }
+        // Sorted so that it's easier to see the diff when this test fails
+        assertEquals(ION_SCHEMA_2_0_SYMBOLS.sorted(), symbols.sorted())
+    }
+
+    @Test
+    fun `getSymbolsTextForSchema should get all expected symbols in a schema`() {
+        val schema = """
+            ${'$'}ion_schema_2_0
+            type::{
+              name: foo_type,
+              annotations: closed::[foo],
```

Comment:
Should we add tests for `annotations` that is defined using [standard syntax](https://amazon-ion.github.io/ion-schema/docs/isl-2-0/spec#standard-syntax)? 
```
annotations: { element: { valid_values: [foo, bar] } }  
```
Also, we might want to explicitly call out in doc comments if we are supporting `annotations` with standard syntax or simple syntax or both.

Reply:
It doesn't really matter, actually. The syntax is different, but the model representation is the same, and the implementation inspects the ISL model rather than the raw ISL.